app/router/router.tsx

```diff
@@ -327,6 +327,18 @@ class Router {
     return Boolean(user?.canCall("listSecrets"));
   }
 
+  getHash() {
+    return window.location.hash.split("@")[0];
+  }
+
+  getLineNumber() {
+    let hashParts = location.hash.split("@");
+    if (hashParts.length > 1) {
+      return parseInt(hashParts[1]);
+    }
+    return undefined;
+  }
```

Comment:
I feel like it's a bit error prone that `window.location.hash` no longer can be trusted, and `router.getHash()` has to be used instead.

wdyt about using a URL param instead? like `?logLine=N`

Reply:
I agree, but I'm conflicted - I almost feel like the selected tab should be in the url since it's something the server might care about (i.e for prefetching and whatever) but the selected line feels like something that truly only the client should care about (which is what belongs in the hash since it's not sent to the server).

Reply:
Would it make more sense if I renamed `getHash()` => `getTab()` and plumbed through "tab" everywhere instead of hash?

Reply:
Yeah, hiding the "hash" terminology is probably a good idea to prevent confusion.

Reply:
Yeah, it's annoying because this is like the one thing that I'm pretty sure belongs in the hash lol. Going to update the hash terminology to tab - I think that prevents a lot of confusion (then you know if you want to convert a hash to a tab, you need to strip a potential line number).